File: components/core/src/test/java/cucumber/runtime/java/StepDefs.java
package cucumber.runtime.java;

import cucumber.annotation.I18n;

import static junit.framework.Assert.assertEquals;

public class StepDefs {
    private String cukes;

    @I18n.EN.Given("^I have (\\d+) cukes in my belly")
    public void haveCukes(String n) {
        this.cukes = n;
    }

    @I18n.EN.Then("^there are (\\d+) cukes in my belly")
    public void checkCukes(String n) {
        assertEquals(cukes, n);
    }

    @I18n.EN.Then("^the (.*) contains (.*)")
    public void containerContainsIngredient(String container, String ingredient) throws InterruptedException {
        assertEquals("glass", container);
    }

    @I18n.EN.Then("^I add (.*)")
    public void addLiquid(String liquid) throws InterruptedException {
        assertEquals("milk", liquid);
    }
}

File: clojure/src/test/resources/cucumber/runtime/clojure/stepdefs.clj
(use 'cucumber.runtime.clojure.belly)

(def some-state (atom "'Before' hasn't run."))

(Before []
  (reset! some-state "'Before' has run.")
  (println "Executing 'Before'."))

(Before ["@foo"]
  (println "Executing 'Tagged Before'"))

(After []
  (println (str "Executing 'After' " @some-state)))

(Given #"^I have (\d+) cukes in my belly$" [cuke-count]
  (eat (Float. cuke-count)))

(Given #"^I have this many cukes in my belly:$" [cuke-table]
  (doseq [x (.raw cuke-table)] (eat (Float. (first x)))))

(When #"^there are (\d+) cukes in my belly$" [expected]
  (assert (= (last-meal) (Float. expected))))

(Then #"^the (.*) contains (.*)$" [container ingredient]
  (assert (= "glass" container)))

(When #"^I add (.*)$" [liquid]
  (assert (= "milk" liquid)))

(Given #"^(\d+) unimplemented step$" [arg1]
  (comment  Express the Regexp above with the code you wish you had  )
  (throw (cucumber.api.PendingException. "This is pending. Seeing a stacktrace here is normal.")))

(def most-recent (atom nil))

(Given #"^I have a kv table:$" [data]
  (reset! most-recent (kv-table->map data)))

(Given #"^I have a table with its keys in a header row:$" [data]
  (reset! most-recent (table->rows data)))

(Then #"^the clojure literal equivalent should be:$" [literal-as-string]
  (assert (= @most-recent (read-string literal-as-string))))
